### PR TITLE
merge dev_crew

### PR DESCRIPTION
**Pull request by USTAADCOM**

### CODE REVIEW

File: apps/automation_crews/dev_crew/agents/backend_developer/backend_developer.py [hunk 1]
```diff
@@ -1,14 +1,17 @@
...
+from dev_crew.tools.file_write import FileWrite
+from dev_crew.tools.file_read import FileRead
 from langchain_groq import ChatGroq
 from dotenv import load_dotenv
+
 load_dotenv()
 
 api_key = os.getenv("GROQ_API_KEY")
-llm = ChatGroq(model = "llama-3.1-70b-versatile", api_key = api_key)
+llm = ChatGroq(model="llama-3.1-70b-versatile", api_key=api_key)
```
Comment (maximiliangarmatsch) [inline apps/automation_crews/dev_crew/agents/backend_developer/backend_developer.py:12]: define this and import from higher up the tree (also in all other places)

File: apps/automation_crews/dev_crew/agents/frontend_developer/frontend_developer.py [hunk 2]
```diff
@@ -1,14 +1,17 @@
...
+from dev_crew.tools.file_write import FileWrite
+from dev_crew.tools.file_read import FileRead
 from langchain_groq import ChatGroq
 from dotenv import load_dotenv
+
 load_dotenv()
 
 api_key = os.getenv("GROQ_API_KEY")
-llm = ChatGroq(model = "llama-3.1-70b-versatile", api_key = api_key)
+llm = ChatGroq(model="llama-3.1-70b-versatile", api_key=api_key)
```
Comment (maximiliangarmatsch) [inline apps/automation_crews/dev_crew/agents/frontend_developer/frontend_developer.py:12]: same

File: apps/automation_crews/dev_crew/agents/project_manager/project_manager.py [hunk 3]
```diff
@@ -1,14 +1,16 @@
...
+from dev_crew.tools.file_write import FileWrite
+from dev_crew.tools.file_read import FileRead
 from langchain_groq import ChatGroq
 from dotenv import load_dotenv
+
 load_dotenv()
 
 api_key = os.getenv("GROQ_API_KEY")
-llm = ChatGroq(model = "llama-3.1-70b-versatile", api_key = api_key)
+llm = ChatGroq(model="llama-3.1-70b-versatile", api_key=api_key)
```
Comment (maximiliangarmatsch) [inline apps/automation_crews/dev_crew/agents/project_manager/project_manager.py:12]: same

File: apps/automation_crews/components/discord/discord_helper_function.py [hunk 4]
```diff
@@ -90,6 +90,30 @@ async def run_assistant_crew(
         print(f"An error occurred while sending a message: {e}")
 
 
+async def run_dev_crew(
+    message: discord.Message, user_message: str, WebsiteDevCrew
+) -> None:
+    if not user_message:
+        print("(Message was empty because intents were not enabled probably)")
+        return
+    is_private = user_message[0] == "?"
```
[the remark this answers is not in the document]
Comment (USTAADCOM) [reply apps/automation_crews/components/discord/discord_helper_function.py:99]: resolved

File: apps/automation_crews/components/discord/discord_helper_function.py [hunk 5]
```diff
@@ -90,6 +90,30 @@ async def run_assistant_crew(
...
+    if is_private:
+        user_message = user_message[1:]
+    try:
+        website_crew = WebsiteDevCrew(project_description=user_message)
+        crew_response = website_crew.run()
+        print(crew_response)
```
[the remark this answers is not in the document]
Comment (USTAADCOM) [reply apps/automation_crews/components/discord/discord_helper_function.py:105]: resolved

File: apps/automation_crews/dev_crew/agents/frontend_developer/frontend_developer.py (hunk 2 above)
Comment (USTAADCOM) [reply apps/automation_crews/dev_crew/agents/frontend_developer/frontend_developer.py]: initialize llm in new created module llm.py and import in rest of the modules.

File: apps/automation_crews/dev_crew/agents/project_manager/project_manager.py (hunk 3 above)
Comment (USTAADCOM) [reply apps/automation_crews/dev_crew/agents/project_manager/project_manager.py]: done

**Review by maximiliangarmatsch — APPROVED** [review]

**Pull request merged**